op-node/rollup/driver/state.go

```diff
@@ -443,3 +456,49 @@ type hashAndErrorChannel struct {
 	hash common.Hash
 	err  chan error
 }
+
+// checkForGapInUnsafeQueue checks if there is a gap in the unsafe queue and attempts to retrieve the missing payloads from the backup RPC.
+// WARNING: This function fails silently (aside from warning logs).
+func (s *Driver) checkForGapInUnsafeQueue(ctx context.Context) {
+	start, end := s.derivation.GetUnsafeQueueGap()
+	size := end - start
+
+	// If there is a gap in the queue and a backup sync client is configured, attempt to retrieve the missing payloads from the backup RPC
+	if size > 0 && s.l2SyncCl != nil {
+		// Attempt to fetch the missing payloads from the backup unsafe sync RPC concurrently.
```

Comment:
Everything inside the if statement should be async. I'd also consider removing the `fetchUnsafeBlockFromRpc` from the driver package. Another option for async communication which is more bounded is to send on channels rather than spawning go routines.

Reply:
Sounds good- @protolambda recommended that we move the logic that concurrently fetches a range of blocks into a function within the `SyncClient` that accepts the `unsafeL2Payloads` channel and can feed them in as they are received, which should be a cleaner place to put the loop + logic in `fetchUnsafeBlockFromRpc`.